stacker/actions/build.py

```diff
@@ -253,10 +253,10 @@ def _get_dependencies(self):
             dependencies[stack.fqn] = stack.requires
         return dependencies
 
-    def pre_run(self, outline=False, *args, **kwargs):
+    def pre_run(self, outline=False, dump=False, *args, **kwargs):
         """Any steps that need to be taken prior to running the action."""
         pre_build = self.context.config.get("pre_build")
-        if not outline and pre_build:
+        if not outline and not dump and pre_build:
```

Comment:
This logic is getting a little long, should we maybe make this a method? `should_run_hooks` or something?